src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package com.dev.cinema.service.impl;
+
+import com.dev.cinema.dao.OrderDao;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.Order;
+import com.dev.cinema.model.Ticket;
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.OrderService;
+import com.dev.cinema.service.ShoppingCartService;
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+
+    @Inject
+    private ShoppingCartService cartService;
+
+    @Override
+    public Order completeOrder(List<Ticket> tickets, User user) {
+        Order order = new Order();
+        order.setUser(user);
+        order.setTickets(tickets);
+        order.setOrderDate(LocalDateTime.now());
+        cartService.clear(cartService.getByUser(user));
+        return orderDao.add(order);
```

Comment:
orderDao.add() will create a delete statement too for shoppingCart tickets. So let's use deep cloning for this list

Reply:
Fixed